lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
     {
-        if (is_array($options_or_dsn)) {
-            $options = array_merge($options_or_dsn, $options);
-        }
+        $this->config = $config;
+        $this->context = new Context();
+        $this->breadcrumbs = new Breadcrumbs();
+        $this->transaction = new TransactionStack();
+        $this->serializer = new Serializer($this->mb_detect_order);
+        $this->reprSerializer = new ReprSerializer($this->mb_detect_order);
+        $this->processors = $this->createProcessors();
 
-        if (!is_array($options_or_dsn) && !empty($options_or_dsn)) {
-            $dsn = $options_or_dsn;
-        } elseif (!empty($_SERVER['SENTRY_DSN'])) {
-            $dsn = @$_SERVER['SENTRY_DSN'];
-        } elseif (!empty($options['dsn'])) {
-            $dsn = $options['dsn'];
-        } else {
-            $dsn = null;
+        if (static::is_http_request() && isset($_SERVER['PATH_INFO'])) {
+            $this->transaction->push($_SERVER['PATH_INFO']);
         }
 
-        if (!empty($dsn)) {
-            $options = array_merge($options, self::parseDSN($dsn));
-        }
-        unset($dsn);
-        $this->init_with_options($options);
+        /*$this->init_with_options($options);
```

Comment:
@nokitakaze I can not say it everywhere you commented, but please read the PR description: this PR is far from being finished, so any review at this point is useless